src/components/TableOfContents/index.tsx

```diff
@@ -56,11 +62,27 @@ function ToCItem<T, K extends string | number>(props: ToCItemProps<T, K>) {
 
     const handleClick = useCallback(
         () => {
-            if (isDefined(key)) {
-                onChange({ key, id });
+            if (isNotDefined(key)) {
+                return;
+            }
+            const isSelected = value.some(s => idSelector(s) === idSelector(option));
+            if (isSelected) {
+                if (multiple) {
+                    const newSelection = value.filter(s => idSelector(s) !== idSelector(option));
+                    onChange(newSelection);
+                } else {
+                    onChange([]);
+                }
```

Comment:
We can use ternary operator here as well
it's more readable that way. but we can also not do it for this case.